backend/dataall/modules/metadata_forms/db/metadata_form_repository.py

```diff
@@ -247,3 +247,27 @@ def query_attached_metadata_forms(session, is_da_admin, groups, user_envs_uris,
         if filter and filter.get('metadataFormUri'):
             query = query.filter(AttachedMetadataForm.metadataFormUri == filter.get('metadataFormUri'))
         return query
+
+    @staticmethod
+    def get_all_attached_metadata_forms_for_entity(session, entityUri, entityType):
+        return (
+            session.query(AttachedMetadataForm)
+            .filter(and_(AttachedMetadataForm.entityType == entityType, AttachedMetadataForm.entityUri == entityUri))
+            .all()
+        )
+
+    @staticmethod
+    def delete_attached_entity_metadata_forms(session, entityUri, entityType):
+        mfs = MetadataFormRepository.get_all_attached_metadata_forms_for_entity(session, entityUri, entityType)
+        for mf in mfs:
+            session.delete(mf)
+
+    @staticmethod
+    def delete_all_home_metadata_forms(session, homeEntityUri, visibility):
+        mfs = (
+            session.query(MetadataForm)
+            .filter(and_(MetadataForm.homeEntity == homeEntityUri, MetadataForm.visibility == visibility))
+            .all()
+        )
+        for mf in mfs:
+            session.delete(mf)
```

Comment:
Instead of querying and then looping and deleting items one by one you could use `delete` directly in the query as a DELETE sql statement:


Like here: 
```
session.query(DatasetTableDataFilter).filter(
            and_(
                DatasetTableDataFilter.tableUri == table.tableUri,
            )
        ).delete()
```

Reply:
It's not used now, but I refactored it anyway

Reply:
If not used let's remove it

Reply:
removed